src/styleguide/template.html

```diff
@@ -87,5 +87,36 @@
     </div></div>
 
     <script src="assets/js/app.js"></script>
+<script type="text/javascript">
+$(function() {
+    var colorsNames = [];
+    $("[data-color-block]").each(function() {
+        var code = rgbToHex($(this).css("background-color"));
+        var name = $(this).data("color-block");
+        colorsNames.push({
+            "name": name,
+            "code": code
+        });
+    });
+    $("[data-color-block-name]").each(function(i) {
+        var name = $(this).data("color-block-name");
+        if(name == colorsNames[i].name){
+            $(this).html(colorsNames[i].code);
+        }
```

Comment:
Currently, it only work because the colors are used in the same order they are declared. You are not only searching for the right color code in `colorsNames`, but only reading for each `[data-color-block-name]` the color code in the order they was declared.

You should rather use:
```js
if (name in colorCodes) {
    $this.html(colorCodes[name]);
}
```

But you can also loop of the current `colorsNames` array, and check for each value if `colorsNames[i].name` matches.